src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -1138,4 +1135,32 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
 
     return recipient ? recipient.recipientName : 'Unknown';
   }
+
+  /**
+  * Logs student activity after student/session details have been fetched.
+  */
+  logStudentAccess(): void {
```

Comment:
Because we need to wait for `loadPersonName` and `loadFeedbackSession` to finish before we can log. The current solution attempts to log once at `loadPersonName` and once at `loadFeedbackSession`

To prevent logging twice, this method just checks that both have been loaded by checking arbitrary data before calling the log service. Because it waits for arbitrary data in a component it might not be very reusable

But once we switch to SQL fully, we can move the check to the log service since `feedbackSessionId` will no longer be allowed to be null

I tried and couldnt get forkJoin to work, maybe if there are any forkjoin experts or other more elegant methods we can try them

Reply:
I maybe have just figured out why forkJoin didnt work. Turns out there are hot and cold observables and you must call subscribe for the ones in Angular's HttpClient...

So if we need to change it I can try again